### PR TITLE
Improve support for "Summary" and "Description" documentation fields

### PR DESCRIPTION
**Pull request by matthchr**
* Fix bug where Indentation parameter of Template.cs counted text after the indentation too.
 * Update PetStore sample slightly.
 * Add support for "Title" field in JSON entities (which maps onto "Summary" field)

### CODE REVIEW

**Comment by matthchr** [comment]
See first commit for the changes.
See second commit for samples regenerate.
See 3rd commit for test regenerate.

**Comment by brjohnstmsft** [comment]
LGTM

**Comment by matthchr** [comment]
@tbombach @fearthecowboy @markcowl -- When you get a chance can you take a look?

**Comment by matthchr** [comment]
@jianghaolu -- if you want to look at the changes I made to Java comments
@amarzavery -- if you want to look at the changes I made to Node comments
@devigned -- if you want to look at the changes I made to Ruby comments
@lmazuel @annatisch -- if you want to look at the changes I made to Python comments

**Comment by tbombach** [comment]
LGTM. I'll merge at EOD if none of the ppl mentioned above have any issues.

**Comment by lmazuel** [comment]
Hi @matthchr, did you made an example/test of each:
- Summary only
- Description only
- Summary + Description

It seems all the current Swagger in tests are "description only". 

Also, it seems that the "description only" scenario does not have the ending "." as we discussed.

File: Samples/petstore/Python/swaggerpetstore/swagger_petstore.py [hunk 1]
```diff
@@ -100,8 +99,10 @@ def add_pet_using_byte_array(
 
     def add_pet(
             self, body=None, custom_headers=None, raw=False, **operation_config):
-        """
-        Add a new pet to the store
+        """Add a new pet to the store
```
Comment (lmazuel) [inline Samples/petstore/Python/swaggerpetstore/swagger_petstore.py]: Ok, found the "description + summary" scenario. Miss the ending "." for the summary.

Comment (matthchr) [inline Samples/petstore/Python/swaggerpetstore/swagger_petstore.py]: Will fix missing "." for summary in Python.  I think all 3 cases above are covered, is there one which is missing?

Comment (lmazuel) [inline Samples/petstore/Python/swaggerpetstore/swagger_petstore.py]: Didn't found the "summary only", sorry :(

Comment (lmazuel) [inline Samples/petstore/Python/swaggerpetstore/swagger_petstore.py]: Yes, I have found them all (y)

**Comment by matthchr** [comment]
@lmazuel Fixed the missing "." bit

**Comment by lmazuel** [comment]
:shipit:

**Comment by lmazuel** [comment]
Funny the icon used by the Codeflow plugin when you check "Approved" :)

**Comment by matthchr** [comment]
@tbombach Resolved merge conflicts -- if it looks good can you merge it?

**Comment by tbombach** [comment]
Merged

**Pull request merged**